test/parameter_test.py

```diff
@@ -385,14 +411,71 @@ class Dogs(luigi.Config):
         class CatsWithoutSection(luigi.ConfigWithoutSection):
             n_cats = luigi.IntParameter()
 
-        luigi.run(['--local-scheduler', '--no-lock', '--n-cats', '123', '--Dogs-n-dogs', '456', 'WithDefault'])
+        self.run_and_check(['--n-cats', '123', '--Dogs-n-dogs', '456', 'WithDefault'])
         self.assertEqual(Dogs().n_dogs, 456)
         self.assertEqual(CatsWithoutSection().n_cats, 123)
 
-        luigi.run(['--local-scheduler', '--no-lock', 'WithDefault', '--n-cats', '321', '--Dogs-n-dogs', '654'])
+        self.run_and_check(['WithDefault', '--n-cats', '321', '--Dogs-n-dogs', '654'])
         self.assertEqual(Dogs().n_dogs, 654)
         self.assertEqual(CatsWithoutSection().n_cats, 321)
 
+    def test_global_significant_param(self):
+        """ We don't want any kind of global param to be positional """
+        class MyTask(luigi.Task):
+            # This could typically be called "--test-dry-run"
+            x_g1 = luigi.Parameter(default='y', is_global=True, significant=True)
+
+        self.assertRaises(luigi.parameter.UnknownParameterException,
+                          lambda: MyTask('arg'))
+
+    def test_global_insignificant_param(self):
+        """ We don't want any kind of global param to be positional """
+        class MyTask(luigi.Task):
+            # This could typically be "--yarn-pool=development"
+            x_g2 = luigi.Parameter(default='y', is_global=True, significant=False)
+
+        self.assertRaises(luigi.parameter.UnknownParameterException,
+                          lambda: MyTask('arg'))
+
+    def test_mixed_params(self):
+        """ Essentially for what broke in a78338c and was reported in #738 """
+        class MyTask(luigi.Task):
+            # This could typically be "--num-threads=True"
+            x_g3 = luigi.Parameter(default='y', is_global=True)
+            local_param = luigi.Parameter()
+
+        MyTask('setting_local_param')
+
+    def test_contrived_positional_parameter(self):
+        """ The original minimal test case we found for #738 looked like this,
+        though it is in fact pretty contrived. Though it's good to have a test
+        case that is more similar to the real-world use case """
+        class TaskWithOneGlobalParam(luigi.Task):
+            """Subclass the JobTask to mess with positional parameters"""
+            # Before commit https://github.com/spotify/luigi/commit/a78338c77d8d79497a7f89bbfde78ba9877acb36
+            # it was possible to use positional parameters and subclasses if the
+            # superclass parameters where is_global
+            # Now they need to be insignificant:
+            # https://github.com/spotify/luigi/commit/f5c33160573a87486c5f0b7423245a89a6d02e0b
+            contrived_global = luigi.Parameter(default='y', is_global=True)
+
+        class TaskWithOneParam(TaskWithOneGlobalParam):
+            the_one_and_only_param = luigi.Parameter()
```

Comment:
the_one_and_only_positional_param? The task actually does have more than one param, right?